.github/workflows/hil_tests.yml

```diff
@@ -215,27 +215,113 @@ jobs:
           pip3 install git+https://github.com/golioth/python-golioth-tools@v0.6.3
 
       - name: Run tests
+        id: run_test
         run: |
           zephyr/scripts/twister                                                    \
               --platform ${{ matrix.west_board }}                                   \
               -T modules/lib/golioth-firmware-sdk/examples/zephyr                   \
+              -C --coverage-basedir modules/lib/golioth-firmware-sdk                \
+              --coverage-formats txt                                                \
               -x=CONFIG_GOLIOTH_COAP_HOST_URI=\"${{ inputs.coap_gateway_url }}\"    \
               --pytest-args="--api-url=${{ inputs.api-url }}"                       \
               --pytest-args="--api-key=${{ secrets[inputs.api-key-id] }}"           \
               --pytest-args="--mask-secrets"
 
+          # Generate coverage summary and format into markdown table
+
+          gcovr -r modules/lib/golioth-firmware-sdk              \
+              --add-tracefile twister-out/coverage.json          \
+              --merge-mode-functions=separate                    \
+              --txt-summary                                      \
+              --html-details twister-out/coverage/index.html     \
+              --xml twister-out/coverage/coverage.xml            \
+              -e "modules/lib/golioth-firmware-sdk/examples/.*"  \
+              -e "modules/lib/golioth-firmware-sdk/external/.*"
+
+          # Echo coverage summary into GitHub Output, with title
+
+          echo "coverage_summary<<EOF" >> $GITHUB_OUTPUT
+          echo "# Code Coverage (${{ matrix.west_board }})" >> $GITHUB_OUTPUT
+          echo "| Type | Coverage |" >> $GITHUB_OUTPUT
+          echo "| --- | --- |" >> $GITHUB_OUTPUT
+          echo "$coverage_summary" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
```

Comment:
I don't think this is actually doing anything, so we can remove it.

Reply:
That is right. It is a leftover from previous attempt at using the same approach as Linux HIL tests.